Neos.Flow/Classes/Security/Authentication/Token/UsernamePasswordInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+namespace Neos\Flow\Security\Authentication\Token;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Mvc\ActionRequest;
+
+/**
+ * Marker interface for authentication tokens which don't need a session.
+ */
+interface UsernamePasswordInterface
+{
+    public function extractCredentialsFromRequest(ActionRequest $request): array;
```

Comment:
The interface should be
```php
public function getUsername(): string;

public function getPassword(): string;
```

IMO since this is not about the extraction mechanism (which might not even be related to the `ActionRequest`) but about the fact that tokens implementing it, hold username & password

Reply:
Well, that also makes sense if we keep it as a `UsernamePasswordInterface` :D

Reply:
Will you make the action request available as a property for the entire Token?

Reply:
I got the idea know. New version pushed